### ISSUE TITLE
Decoding bug: nested atom wrongfully parsed to nil

### ISSUE THREAD

**Issue by dimitarvp** [gaynetdinov/ex_marshal#19]
Hello, when working to decrypt and decode Rails 4.0 sessions, I stumbled upon a bug. Long story short, get in `iex` and follow this:

```elixir
marshalled = "\x04\b{\tI\"\x0Fsession_id\x06:\x06ETI\"%9abcec93dd746529e91623edaa51a69f\x06;\x00TI\"\x1Fwarden.user.spree_user.key\x06;\x00T[\a[\x06i\x02\xC0{I\"\x19bvxmDzqkuhesXSxn85_6\x06;\x00TI\"\nflash\x06;\x00T{\aI\"\fdiscard\x06;\x00T[\x06:\fsuccessI\"\fflashes\x06;\x00T{\x06;\x06I\"\eLogged in successfully\x06;\x00TI\"\x10_csrf_token\x06;\x00FI\"1RlmOff4/q/+Pb2JhNObj1uJkPe7vRwi/BOHB34NYZ84=\x06;\x00F"

actual = ExMarshal.decode(marshalled)
=> %{"_csrf_token" => "RlmOff4/q/+Pb2JhNObj1uJkPe7vRwi/BOHB34NYZ84=",
  "flash" => %{"discard" => [:success],
    "flashes" => %{nil: "Logged in successfully"}},
  "session_id" => "9abcec93dd746529e91623edaa51a69f",
  "warden.user.spree_user.key" => [[31680], "bvxmDzqkuhesXSxn85_6"]}

# This has been obtained via Ruby code: Marshal.load(marshalled), copied in Elixir (added % signs for the maps) and then printed to Elixir console.
expected = %{"_csrf_token" => "RlmOff4/q/+Pb2JhNObj1uJkPe7vRwi/BOHB34NYZ84=",
  "flash" => %{"discard" => [:success],
    "flashes" => %{success: "Logged in successfully"}},
  "session_id" => "9abcec93dd746529e91623edaa51a69f",
  "warden.user.spree_user.key" => [[31680], "bvxmDzqkuhesXSxn85_6"]}

get_in expected, ["flash", "flashes"]                              
=> %{success: "Logged in successfully"}

# This is the problem:
get_in actual, ["flash", "flashes"]
=> %{nil: "Logged in successfully"}
```

It seems that this `success` key in the nested map is not parsed properly? The problem persists regardless of the `nullify_objects` config option, too.

**Comment by gaynetdinov** [comment]
Thanks for the bug report. I found the issue and will try to fix it asap. 
The bug is about repetitive symbols. In your map there are two occurrences of `success`. Ruby Marshal encodes repetitive symbols as links to the first ones, so it's a bit tricky to decode.

**Comment by dimitarvp** [comment]
I reckoned it's something like that. I'll be monitoring this issue for
updates.

I appreciate your work!

**Comment by gaynetdinov** [comment]
@dimitarvp I believe I fixed the issue you can check PR I referenced above. I'd suggest you to try `consider-encoding-symbol` branch first and if it works for you, I'll merge this PR and will release new version.

**Comment by dimitarvp** [comment]
Using this:

```elixir
{:ex_marshal, github: "gaynetdinov/ex_marshal", branch: "consider-encoding-symbol"}
```

in my `mix.exs` fixed the issue indeed. Thank you!

Waiting for your next official release.

**Comment by gaynetdinov** [comment]
Sorry for the delay, I just released v0.0.10 https://hex.pm/packages/ex_marshal/0.0.10

**Comment by dimitarvp** [comment]
@gaynetdinov Just tested manually, works perfect on 0.0.10 as well.

Thanks!
